### ISSUE TITLE
Cannot use $default WebSocket route again

### ISSUE THREAD

**Issue by krlozadan** [serverless/serverless#6507]
# This is a Bug Report

## Description

- What went wrong?

I removed the $default route function definition from my serverless.yml file and deployed it correctly. After, I added the $default route again and it won't let me deploy since it says the route already exists. I've tried renaming the function, moving the $default route event to an existing function. But nothing works. Apparently even if you remove the $default route it will still create it.

- What did you expect should have happened?

Add/replace the $default route

- What was the config you used?

- What stacktrace or error message from your provider did you see?
`An error occurred: SdefaultWebsocketsRoute - Route with key $default already exists for this API (Service: AmazonApiGatewayV2; Status Code: 409; Error Code: ConflictException; Request ID: xxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx)`

Similar or dependent issues:

- N/A

## Additional Data

- **_Serverless Framework Version you're using_**: 1.47.0
- **_Operating System_**: macOS Mojave v10.14.5 running Node 10.15.3
- **_Stack Trace_**: See Above
- **_Provider Error messages_**: See Above

**Comment by krlozadan** [comment]
Hi, any status update on this? I'd like to try solving this if possible

**Comment by yufeng0924** [comment]
are you solve it already?

**Comment by rpaulojade** [comment]
Have we got any idea as to how to solve this? Because I've been having the same problem with API Gateway.

**Comment by andresandrade12345** [comment]
Any updates?

**Comment by urthling** [comment]
I seem to be running into the same issue..  any updates?

**Comment by diegomarzo** [comment]
I have been bitten by this issue today and lost a good amount of hours.

My "solution" (in quotes because it is a bit crappy but it worked for me at least).

1) If you have a domain for the sockets in you API Gateway remove it, you can go to API Gateway and remove it from there or using the `sls delete_domain`

2) Comment all the WS functions from your serverless.yml and any reference to them (for example I had this guy: 
```    APIG_ENDPOINT:
      Fn::Join:
        - ""
        - - Ref: WebsocketsApi
          - .execute-api.
          - Ref: AWS::Region
          - .amazonaws.com/
          - ${self:provider.stage}```)

3) Now `sls deploy`, this way serverless will deploy your stack removing the Websocket stuff
4) If you want to have your WS domain, now run `sls create_domain` or create manually
5) Uncomment all your functions and your stuff
6) `sls deploy` now. Everything should work fine, at least it did for me.

I'm happy to have found this post because it sort of pointed me into the good direction, when you are working with SOOOOOOO many things at the same time feels so easy to think that you f*cked up around :D, so I have been commenting functions, rewriting them, changing names...

It is also interesting that I have everything running locally and locally all worked, so at least I knew it was not a code issue.

For the records, this was my error:

`An error occurred: WebsocketsDeployment - A route target (typically an Integration) must be defined for route xg1glns with key $default (Service: AmazonApiGatewayV2; Status Code: 400; Error Code: BadRequestException; Request ID: ab23fdc4-eda0-99aa-2341-abd12928172727; Proxy: null).`

And if I tried to add something called "$default" then it gave me the reporters error.

I hope this help!!!!

**Comment by qmg-rwaller** [comment]
We had the same error after removing the `$default` route and integration from our _vanilla CloudFormation_ - the integration deleted but the route lingered, and subsequent deployments of the stage failed **silently** with that error! I think this bug is in vanilla Cloudformation, though perhaps serverless could implement a workaround

**Comment by semmgeorge** [comment]
how I fix it:

**Comment by simonbuchan** [comment]
TLDR Serverless shouldn't be creating a `$default` route, it should instead use the "api gateway managed overrides" resource: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigatewayv2-apigatewaymanagedoverrides.html
